File: tests/test_cpg/fixtures/parity_xss.java
/** Cross-language XSS fixture — Java (Spring). */

import org.springframework.web.bind.annotation.*;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

@RestController
public class XssController {

    @GetMapping("/hello")
    public void hello(@RequestParam("name") String name,  // $ source=xss
                       HttpServletResponse response) throws IOException {
        response.getWriter().write("<h1>Hello " + name + "</h1>");  // $ sink=xss
    }

    @GetMapping("/reflect")
    public void reflect(@RequestParam("msg") String msg,  // $ source=xss
                        HttpServletResponse response) throws IOException {
        response.getWriter().write("<div>" + msg + "</div>");  // $ sink=xss
    }
}

File: tests/test_cpg/fixtures/parity_xss.py
"""Cross-language XSS fixture — Python (Flask)."""

from flask import Flask, Response, request

app = Flask(__name__)


@app.route("/hello")
def hello():
    name = request.args.get("name", "World")  # $ source=xss
    return Response(f"<h1>Hello {name}</h1>")  # $ sink=xss


@app.route("/reflect")
def reflect():
    user_input = request.args.get("msg", "")  # $ source=xss
    return Response("<div>" + user_input + "</div>")  # $ sink=xss
